java-checks/src/main/java/org/sonar/java/checks/LoggedRethrownExceptionsCheck.java

```diff
@@ -0,0 +1,122 @@
+    MethodMatcher.create().typeDefinition(JAVA_UTIL_LOGGING_LOGGER).name("severe").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(JAVA_UTIL_LOGGING_LOGGER).name("warning").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(SLF4J_LOGGER).name("debug").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(SLF4J_LOGGER).name("error").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(SLF4J_LOGGER).name("info").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(SLF4J_LOGGER).name("trace").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(SLF4J_LOGGER).name("warn").withAnyParameters()
+  ));
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.CATCH);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (!hasSemantic()) {
+      return;
+    }
+    CatchTree catchTree = (CatchTree) tree;
+    boolean isLogging = false;
+    List<Location> secondaryLocations = new ArrayList<>();
+    for (StatementTree statementTree : catchTree.block().body()) {
+      if (isLogging && statementTree.is(Tree.Kind.THROW_STATEMENT)) {
+        secondaryLocations.add(new Location("", ((ThrowStatementTree) statementTree).expression()));
+        reportIssue(catchTree.parameter(), "Either log this exception and handle it, or rethrow it with some contextual information.", secondaryLocations, 0);
+        return;
+      } else if (isLoggingMethod(statementTree, catchTree.parameter().simpleName())) {
+        secondaryLocations.add(new Location("", statementTree));
+        isLogging = true;
+      }
+    }
+  }
+
+  private static boolean isLoggingMethod(StatementTree statementTree, IdentifierTree exceptionIdentifier) {
+    if (!statementTree.is(Tree.Kind.EXPRESSION_STATEMENT)) {
+      return false;
+    }
+    ExpressionTree expression = ((ExpressionStatementTree) statementTree).expression();
+    if (expression.is(Tree.Kind.METHOD_INVOCATION)) {
+      MethodInvocationTree mit = (MethodInvocationTree) expression;
+      return LOGGING_METHODS.stream().anyMatch(logMethod -> logMethod.matches(mit)) && isExceptionUsed(exceptionIdentifier, mit);
+    }
+    return false;
+  }
+
+  private static boolean isExceptionUsed(IdentifierTree exceptionIdentifier, MethodInvocationTree mit) {
+    ExceptionUsageVisitor visitor = new ExceptionUsageVisitor(exceptionIdentifier);
+    mit.arguments().forEach(param -> param.accept(visitor));
+    return visitor.isExceptionIdentifierUsed;
+  }
+
+
+  private static class ExceptionUsageVisitor extends BaseTreeVisitor {
+
+    IdentifierTree exceptionIdentifier;
+    boolean isExceptionIdentifierUsed = false;
+
+
+    ExceptionUsageVisitor(IdentifierTree exceptionIdentifier) {
+      this.exceptionIdentifier = exceptionIdentifier;
+    }
+
+    @Override
+    public void visitIdentifier(IdentifierTree tree) {
+      if (tree.name().equals(exceptionIdentifier.name())) {
```

Comment:
There is no need to recheck equality if `isExceptionIdentifierUsed` is already true